### ISSUE TITLE
upload error

### ISSUE THREAD

**Issue by HansVissers** [stanstrup/PredRet#80]
Hello, The attached csv table creates an upload error. Thank you for looking into this, Hans

[FDAtrain4PredRet.zip](https://github.com/stanstrup/PredRet/files/4121680/FDAtrain4PredRet.zip)

![1-28-2020 10-44-03 AM](https://user-images.githubusercontent.com/33374753/73252640-54770f00-41bb-11ea-8359-b1755a49d75a.jpg)

**Comment by stanstrup** [comment]
Thanks for the report. The file appears to be correctly formatted.

Could you try making the pubchem column empty (but keeping the column)? The pull from pubchem doesn't work anymore. I don't remember if it uses that even if there is an inchi.

Could you also set the Method to "RPMMFDA" instead of "RPMM" so there is no confusion?

Neither of these things should make it break of course. But the system is not the most robust...

**Comment by HansVissers** [comment]
Dear Jan,

Thank you for the very prompt reply.

I initially tried to upload/process the file without the PubChem CIDs but this resulted in the same error. The method is set to "RPMMFDA".

Please let me know if you need any additional information for trouble shooting.

Best regards,

Hans

Hans Vissers
Senior Consulting Scientist | Scientific Operations
Waters Corporation

phone +44 161 435 4100
web www.waters.com<http://www.waters.com/>     email hans_vissers@waters.com<mailto:hans_vissers@waters.com>
Stamford Avenue | Altrincham Road, Wilmslow, SK9 4AX UK

Micromass UK Limited is registered in England under No 3162904
Registered Office: Stamford Avenue | Altrincham Road, Wilmslow, SK9 4AX

[cid:image001.gif@01D5D5CC.76636DD0]

**Comment by stanstrup** [comment]
Yes you have selected that method but the file says "RPMM". As far as I remember what is in the CSV file overrules. Could you try changing that in the CSV?

**Comment by HansVissers** [comment]
Dear Jan,

Sorry, I misunderstood. I updated the csv file but I got the same error. The console reads the following:

JQMIGRATE: Migrate is installed, version 1.4.0
(index):113 Uncaught TypeError: Cannot set property 'value' of null
    at (index):113
(index):347 The specified value "NA" is not a valid number. The value must match to the following regular expression: -?(\d+|\d+\.\d+|\.\d+)([eE][-+]?\d+)?
(index):356 The specified value "NA" is not a valid number. The value must match to the following regular expression: -?(\d+|\d+\.\d+|\.\d+)([eE][-+]?\d+)?
(index):367 The specified value "NA" is not a valid number. The value must match to the following regular expression: -?(\d+|\d+\.\d+|\.\d+)([eE][-+]?\d+)?
(index):376 The specified value "NA" is not a valid number. The value must match to the following regular expression: -?(\d+|\d+\.\d+|\.\d+)([eE][-+]?\d+)?
(index):247 Uncaught TypeError: $ is not a function
    at Object.resizedCallback ((index):247)
    at a (iframeResizer.min.js:8)
sockjs-0.3.min.js:27 WebSocket connection to 'ws://predret.org/shiny/shiny.apps/apps/retpred_shiny/retdb/__sockjs__/520/kp4pbdbz/websocket' failed: Error during WebSocket handshake: Unexpected response code: 400
y.websocket @ sockjs-0.3.min.js:27
2(index):247 Uncaught TypeError: $ is not a function
    at Object.resizedCallback ((index):247)
    at a (iframeResizer.min.js:8)
shiny-server.js:210
Listening on http://127.0.0.1:38485
Loading required package: rJava
Loading required package: xlsxjars
Loading required package: Rcpp
Warning: 'loadRcppModules' is deprecated.
Use 'loadModule' instead.
See help("Deprecated")
Warning: replacing previous import ‘xcms::groups’ by ‘igraph::groups’ when loading ‘CAMERA’
Loading required package: RcppClassic
Loading required package: ggplot2

Attaching package: ‘DT’

The following objects are masked from ‘package:shiny’:

    dataTableOutput, renderDataTable

Warning: Unhandled error in observer: error in running command
observe({
    if (is.null(input$upload_go_Button))
        return(NULL)
    if ((input$upload_go_Button) == 0)
        return(NULL)
    if (is.null(input$files))
        return(NULL)
    if (any(unlist(lapply(data_cleaned()$errors, function(x) x$error ==
        1))))
        return(NULL)
    isolate({
        if (is.null(data_has_been_written$done)) {
            data_has_been_written$done <- 0L
        }
        else {
            data_has_been_written$done <- 0L
        }
        bson_data = mongo.bson.from.df(data_cleaned()$data)
        mongo <- PredRet_connect()
        wrote = mongo.insert.batch(mongo, PredRet.env$namespaces$ns_rtdata,
            bson_data)
        del <- mongo.disconnect(mongo)
        del <- mongo.destroy(mongo)
        data_has_been_written$done <- as.numeric(wrote)
    })
})

Best regards,

Hans

Hans Vissers
Senior Consulting Scientist | Scientific Operations
Waters Corporation

phone +44 161 435 4100
web www.waters.com<http://www.waters.com/>     email hans_vissers@waters.com<mailto:hans_vissers@waters.com>
Stamford Avenue | Altrincham Road, Wilmslow, SK9 4AX UK

Micromass UK Limited is registered in England under No 3162904
Registered Office: Stamford Avenue | Altrincham Road, Wilmslow, SK9 4AX

[cid:image001.gif@01D5D5D2.A5569120]

**Comment by stanstrup** [comment]
Sorry for the goose chases.

I think I found the bug. At least I was able to crash it. The system uses obabel to clean up the structures and there is a limit to the length of the command that can be send. One of my functions is missing a safeguard for that apparently.

Can you try to dividing your file in two of approximately similar size? That solves _that_ problem here.

**Comment by HansVissers** [comment]
Dear Jan,

Thanks for looking into this.

I will give that a try. Can I inform the software that 2 uploads need to be used to create a model? Thanks.

Best regards,

Hans

Hans Vissers
Senior Consulting Scientist | Scientific Operations
Waters Corporation

phone +44 161 435 4100
web www.waters.com<http://www.waters.com/>     email hans_vissers@waters.com<mailto:hans_vissers@waters.com>
Stamford Avenue | Altrincham Road, Wilmslow, SK9 4AX UK

Micromass UK Limited is registered in England under No 3162904
Registered Office: Stamford Avenue | Altrincham Road, Wilmslow, SK9 4AX

[cid:image001.gif@01D5D5E0.9B8DA6C0]

**Comment by stanstrup** [comment]
As long as they have the same "Method" all data will be used.
If you are unlucky it makes the model based on the first set you upload. Then it will redo the model with all the data on the second pass. The system looks for new data every 10 minutes. It can take longer than that to finish the modeling though. If you give it a few hours it should be plenty to get all predictions possible.

**Comment by HansVissers** [comment]
On its way:

[cid:image001.png@01D5D5E2.3EC6D180]

**Comment by stanstrup** [comment]
your image didn't survive email --> github. Can you put on github directly?

**Comment by HansVissers** [comment]
![1-28-2020 2-28-10 PM](https://user-images.githubusercontent.com/33374753/73268008-70d67400-41da-11ea-874f-3a7b92539220.jpg)

Here it is!

**Comment by stanstrup** [comment]
Ah. So it did work!

All good?

**Comment by HansVissers** [comment]
Yes, I assume that the “system” will become available once the model has been created. Thanks.

**Comment by stanstrup** [comment]
A thousand predictions now available ;)

I will leave this open as the underlying bug is not fixed.

**Comment by YunyunZHU** [comment]
Hello, stanstrup,

I've tried to split my file into three parts, each has about 15 compounds. And I have Pubchem column even if they are all empty. But I still get the error message identical to HansVissers. I'm wondering what else I can do to solve this? Looking forward to hearing from you!
Thank you!
